pkg/connection/store/kubernetes/store_test.go

```diff
@@ -118,6 +118,22 @@ func TestSecretStoreReadKeyValues(t *testing.T) {
 				result: store.KeyValues(fakeKV()),
 			},
 		},
+		"SecretNotFound": {
+			reason: "Should return an empty secret and without any error if the secret is not found",
+			args: args{
+				client: resource.ClientApplicator{
+					Client: &test.MockClient{
+						MockGet: test.NewMockGetFn(kerrors.NewNotFound(schema.GroupResource{}, "")),
```

Comment:
We should return error only if we try to retrieve the secret with the name equal to `fakeSecretName`. Otherwise, if we do not perform this check, this tests is going to pass even if `ReadKeyValues` does not properly use the passed `ScopedName`

Reply:
yeah, that's the issue with mocking, the mock is never going to be 100% precise with the real behaviour. I don't think it's strictly required here, but could be a separate test case maybe.

Reply:
mocks should react for only specific inputs. We are adding here the new test, so why not making it more tight here?

Reply:
+1 for adding that to the SuccessfulRead test.

Reply:
Actually, probably the right thing to do would be to add a test case getting a specific secret and returning an errBoom error only for the specified name. That would ensure we are actually calling it with the right name.

Reply:
> The NewMockGetFn function does not use the object key. Which means we cannot check the scope name with that.

yeah, but we can do something like:

```go
Client: &test.MockClient{
  MockGet: func(ctx context.Context, key client.ObjectKey, obj client.Object) error {
	if key.Name == fakeSecretName {
		return kerrors.NewNotFound(schema.GroupResource{}, "")
	}
	return errors.Errorf("Should return not found error actually for %s", key)
},
},
```

Reply:
I know we can. But my point was, `NewMockGetFn` function is used for dozens of places without actually checking the name or namespace and I think it's OK. IMHO, if we're mocking an interface, we don't need to worry about the mocker itself. The critical part is checking the behavior after getting the result from the mock. 

Anyways, I updated the `SuccessfulRead` test.

Reply:
> I know we can. But my point was, NewMockGetFn function is used for dozens of places without actually checking the name or namespace and I think it's OK. IMHO, if we're mocking an interface, we don't need to worry about the mocker itself. The critical part is checking the behavior after getting the result from the mock.

We are not mocking just interface, we are actually mocking its invocations, and we in essence should define a number of invocations as set of (input params, results).

The code under tests should invoke an interface with proper input params, and should react properly on the obtained results. If we do not check the input params, then we verify just partially the code behavior.